### PR TITLE
SR-IOV: support querying state from Nispor

### LINKED ISSUE #1300: Use nispor for run time SR-IOV status query

### PR DESCRIPTION
**Pull request by ffmancera**
Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

### CODE REVIEW

**Comment by ffmancera** [comment]
This is waiting https://github.com/nispor/nispor/pull/36

**Comment by ffmancera** [comment]
This is waiting for a new release on Nispor.

**Comment by ffmancera** [comment]
Tested on SR-IOV hardware.

File: libnmstate/nispor/ethernet.py [hunk 1]
```diff
@@ -26,3 +27,28 @@ class NisporPluginEthernetIface(NisporPluginBaseIface):
...
+            for vf in self.np_iface.sr_iov.vfs:
+                vf_info = {
+                    Ethernet.SRIOV.VFS.ID: vf.vf_id,
+                    Ethernet.SRIOV.VFS.MAC_ADDRESS: vf.mac,
+                    Ethernet.SRIOV.VFS.SPOOF_CHECK: vf.spoof_check,
+                    Ethernet.SRIOV.VFS.TRUST: vf.trust,
+                    Ethernet.SRIOV.VFS.MIN_TX_RATE: vf.min_tx_rate,
+                    Ethernet.SRIOV.VFS.MAX_TX_RATE: vf.max_tx_rate,
+                }
+                vf_infos.append(vf_info)
```
Comment (cathay4t) [inline libnmstate/nispor/ethernet.py]: This can be simplified by `vf_infos.append({...})`

File: libnmstate/nispor/ethernet.py [hunk 2]
```diff
@@ -26,3 +27,28 @@ class NisporPluginEthernetIface(NisporPluginBaseIface):
...
+
+            sriov_info = {
+                Ethernet.SRIOV.TOTAL_VFS: len(self.np_iface.sr_iov.vfs),
+                Ethernet.SRIOV.VFS_SUBTREE: vf_infos,
+            }
+            info[Ethernet.CONFIG_SUBTREE] = {
```
Comment (cathay4t) [inline libnmstate/nispor/ethernet.py:51]: This is incorrect according to `schemas/operational-state.yaml`.
The schema file put `sr-iov` at the top level of `iface_info`.
While the `nm/sriov.py` use `Ethernet.CONFIG_SUBTREE`.

Please create a separate PR fixing this first.

**Review by cathay4t — CHANGES_REQUESTED** [review]

File: libnmstate/nispor/ethernet.py (hunk 2 above)
Comment (ffmancera) [reply libnmstate/nispor/ethernet.py:51]: Done on #1323

File: libnmstate/nispor/ethernet.py (hunk 1 above)
Comment (ffmancera) [reply libnmstate/nispor/ethernet.py]: Done, thanks!

**Review by cathay4t — APPROVED** [review]

**Comment by ffmancera** [comment]
Testing it again we the need changes, please do not merge. I will do after testing is successfully. Thanks

**Pull request merged**